pkg/types/namespacedname.go

```diff
@@ -0,0 +1,58 @@
+/*
+Copyright 2015 Google Inc. All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package types
+
+// NamespacedName comprises a resource name, with a mandatory namespace,
+// rendered as "<namespace>/<name>".  Being a type captures intent and
+// helps make sure that UIDs, namespaced names and non-namespaced names
+// do not get conflated in code.  It also provides a convenient place for
+// utility functions to do things like splitting, joining and rendering.
+type NamespacedName struct {
+	namespace string
+	name      string
+}
+
+// NewNamespacedName is the constructor
+func NewNamespacedName(namespace, name string) *NamespacedName {
```

Comment:
I think we'll have to agree to differ on this point. I've added string validation to NewNamespacedName(), and that highlighted a bunch more bugs in pkg/proxy/config/config_test.go that I've now fixed too.  Hopefully that
provides some support for my pedantic encapsulation and string validation :-)

As an aside, to avoid circular imports, I've moved NamespacedName to pkg/types/valid.  As an aside it makes for slightly nicer-to-read code.